Tools/TestWebKitAPI/Tests/WebKitCocoa/ProcessSwapOnNavigation.mm

```diff
@@ -7165,6 +7165,191 @@ static unsigned waitUntilScrollPositionIsRestored(WKWebView *webView)
     done = false;
 }
 
+static const char* beforeUnloadPageBytes = R"PSONRESOURCE(
+<!DOCTYPE html>
+<html>
+<body>
+<script>
+window.addEventListener('beforeunload', (event) => {
+    event.preventDefault();
+    if (window.webkit && window.webkit.messageHandlers && window.webkit.messageHandlers.pson)
+        window.webkit.messageHandlers.pson.postMessage("DidFireBeforeUnload");
+    return event.returnValue = "Are you sure you want to exit?";
+});
+</script>
+</body>
+</html>
+)PSONRESOURCE";
+
+static const char* beforeUnloadFinishPageBytes = R"PSONRESOURCE(
+<!DOCTYPE html>
+<html>
+<body>
+</body>
+</html>
+)PSONRESOURCE";
+
+static bool shouldRejectClosingViaPrompt = false;
+static bool didRespondToPrompt = false;
+
+@interface BeforeUnloadPromptUIDelegate : NSObject <WKUIDelegate>
+@end
+
+@implementation BeforeUnloadPromptUIDelegate
+
+- (void)_webView:(WKWebView *)webView runBeforeUnloadConfirmPanelWithMessage:(NSString *)message initiatedByFrame:(WKFrameInfo *)frame completionHandler:(void (^)(BOOL result))completionHandler
+{
+    dispatch_after(dispatch_time(DISPATCH_TIME_NOW, 0.5 * NSEC_PER_SEC), dispatch_get_main_queue(), ^{
+        completionHandler(shouldRejectClosingViaPrompt ? NO : YES);
+        didRespondToPrompt = true;
+    });
+}
+
+@end
+
+TEST(ProcessSwap, BeforeUnloadModalAlert)
+{
+    auto processPoolConfiguration = psonProcessPoolConfiguration();
+    auto processPool = adoptNS([[WKProcessPool alloc] _initWithConfiguration:processPoolConfiguration.get()]);
+
+    auto webViewConfiguration = adoptNS([[WKWebViewConfiguration alloc] init]);
```

Comment:
nit: I like to use new instead of alloc init because it's more concise.  Is it important? No.